Library/Homebrew/dev-cmd/boneyard-formula-pr.rb

```diff
@@ -0,0 +1,125 @@
+      end
+      tap_migrations = Utils::JSON.load(File.read(formula.tap.path/"tap_migrations.json"))
+      tap_migrations[formula.name] = "homebrew/boneyard"
+      tap_migrations = tap_migrations.sort.inject({}) { |a, e| a.merge!(e[0] => e[1]) }
+      (formula.tap.path/"tap_migrations.json").atomic_write(JSON.pretty_generate(tap_migrations) + "\n")
+    end
+    unless Formula["hub"].any_version_installed?
+      if ARGV.dry_run?
+        puts "#{HOMEBREW_BREW_FILE} install hub"
+      else
+        safe_system HOMEBREW_BREW_FILE, "install", "hub"
+      end
+    end
+    branch = "#{formula.name}-boneyard"
+    puts "cd #{formula.tap.formula_dir}" if ARGV.dry_run?
+    cd formula.tap.formula_dir
+    if ARGV.dry_run?
+      puts "git checkout -b #{branch} origin/master"
+      puts "git commit --no-edit --verbose --message=\"#{formula.name}: migrate to boneyard\" -- #{formula_path} #{formula.tap.path}/tap_migrations.json"
+      puts "hub fork --no-remote"
+      puts "hub fork"
+      puts "hub fork (to read $HUB_REMOTE)"
+      puts "git push $HUB_REMOTE #{branch}:#{branch}"
+      puts "hub pull-request -m $'#{formula.name}: migrate to boneyard\\n\\nCreated with `brew boneyard-formula-pr`.'"
+    else
+      safe_system "git", "checkout", "-b", branch, "origin/master"
+      safe_system "git", "commit", "--no-edit", "--verbose",
+        "--message=#{formula.name}: migrate to boneyard",
+        "--", formula_path, formula.tap.path/"tap_migrations.json"
+      safe_system "hub", "fork", "--no-remote"
+      quiet_system "hub", "fork"
+      remote = Utils.popen_read("hub fork 2>&1")[/fatal: remote (.+) already exists./, 1]
+      odie "cannot get remote from 'hub'!" unless remote
+      safe_system "git", "push", remote, "#{branch}:#{branch}"
+      safe_system "hub", "pull-request", "-m",
+       "#{formula.name}: migrate to boneyard\n\nCreated with `brew boneyard-formula-pr`."
+    end
+    puts "cd #{Tap.fetch("homebrew", "boneyard").formula_dir}" if ARGV.dry_run?
+    cd Tap.fetch("homebrew", "boneyard").formula_dir
+    if ARGV.dry_run?
+      puts "git checkout -b #{branch} origin/master"
+      if bottle_block
+        puts "Removing bottle block"
+      else
+        puts "No bottle block to remove"
+      end
+      puts "git add #{formula.name}.rb"
+      puts "git commit --no-edit --verbose --message=\"#{formula.name}: migrate from #{formula.tap.repo}\" -- #{formula.name}.rb"
+      puts "hub fork --no-remote"
+      puts "hub fork"
+      puts "hub fork (to read $HUB_REMOTE)"
+      puts "git push $HUB_REMOTE #{branch}:#{branch}"
+      puts "hub pull-request -m $'#{formula.name}: migrate from #{formula.tap.repo}\\n\\nCreated with `brew boneyard-formula-pr`.'"
+    else
+      safe_system "git", "checkout", "-b", branch, "origin/master"
+      if bottle_block
+        Utils::Inreplace.inreplace("#{formula.name}.rb") do |s|
+          s.sub!(/  bottle do.+?end\n\n/m, "")
+        end
```

Comment:
Fair enough, as I suppose we're now approaching the point where http://xkcd.com/1691/ no longer applies.